### PR TITLE
Differences between HF and Google Sheets and Excel

### PR DESCRIPTION
**Pull request by kirszenbaum**
* Added the tables from #603 here:

https://github.com/handsontable/hyperformula/blob/feature/issue-603/docs/guide/known-limitations.md

### CODE REVIEW

File: docs/guide/known-limitations.md [hunk 1]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+## Google Sheets and Microsoft Excel
+
+In certain situations, HyperFormula behaves slightly differently than Google Sheets or Microsoft Excel.
+
+The inconsistencies are due to:
+* Known limitations of Microsoft Excel or Google Sheets.
+* Known limitations of HyperFormula in its current development stage.
+* Inconsistencies between Microsoft Excel and Google Sheets.
+
+In most cases, the differences won't be observed by the end user, but we list them all for your reference:
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:46]: Well... we got few issue reports that they did noticed 😉 So I wouldn't agree on that line

Also, the first table is out of their hands. But second one may be fixed by customized implementations of listed functions. So if someone is determined enough, may overwrite our implementations with theirs. IMO worth to mention that those limitations may be removed, we just had to choose one implementation

File: docs/guide/known-limitations.md [hunk 2]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| DAYS          | =DAYS(-1, 0)                                               |          NUM |      -1,0000 |        NUM |
+| DAYS          | =DAYS(0, -1)                                               |          NUM |       1,0000 |        NUM |
+| DATEDIF       | =DATEDIF(-1, 0,   "Y")                                     |          NUM |       0,0000 |        NUM |
+| RATE          | =RATE(12, -100, 400, 100, 1)                               |      -0,4997 |       0,3100 |    -0,4997 |
+| RATE          | =RATE(12, -100, 400, 1, 1)                                 |      -0,9901 |       0,3200 |    -0,9901 |
+| RATE          | =RATE(12, -100, 400, 0, 1)                                 |          NUM |       0,3200 |        NUM |
+| RATE          | =RATE(12, -100, 400, -100, 1)                              |          NUM |       0,3200 |        NUM |
+| RATE          | =RATE(0.9, -100, 400)                                      |      -0,7962 |          NUM |    -0,7962 |
+| LCMP          | =LCM(1000000,1000001,1000002,1000003)                      |          NUM |  5,00003E+23 |        NUM |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:79]: Should break the line. Currently table is wider than page width 
```suggestion
| LCMP          | =LCM(1000000, 1000001, 1000002, 1000003)                   |          NUM |  5,00003E+23 |        NUM |
```

File: docs/guide/known-limitations.md [hunk 3]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| TBILLPRICE    | =TBILLPRICE(0, 180, 1.9)                                   |       5,0000 |          NUM |     5,0000 |
+| TBILLPRICE    | =TBILLPRICE(0, 180, 2)                                     |       0,0000 |          NUM |     0,0000 |
+| NPV           | =NPV(1, TRUE(), 1)                                         |       0,7500 |       0,5000 |     0,7500 |
+| NPV           | =NPV(1,B1) where B1 = true                                 |       0,5000 |       0,0000 |     0,0000 |
+| POISSON.DIST  | =POISSON.DIST(-0.01, 0, FALSE())                           |          NUM |       1,0000 |        NUM |
+| POISSON.DIST  | =POISSON.DIST(0, -0.01, FALSE())                           |          NUM |          NUM |     1,0101 |
+| DB            | =DB(1000000,100000,6,7,7)                                  |   15845,1000 |          NUM | 15845,0985 |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:86]: Just for formatting consistency
```suggestion
| DB            | =DB(1000000, 100000, 6, 7, 7)                              |   15845,1000 |          NUM | 15845,0985 |
```

File: docs/guide/known-limitations.md [hunk 4]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| HYPGEOM.DIST  | =HYPGEOM.DIST(4, 12, 20, 40, TRUE())                       |       0,1504 |          N/A |     0,1504 |
+| TDIST         | =TDIST(0, 1, 1.5)                                          |          NUM |       0,5000 |     0,5000 |
+| T.INV.2T      | =T.INV.2T(0, 1)                                            |          NUM |          NUM |      DIV/0 |
+| T.DIST        | =T.DIST(1, 0.9, FALSE())                                   |          NUM |          NUM |      DIV/0 |
+| AVEDEV        | =AVEDEV(TRUE(),FALSE(),)                                   |       0,4444 |       0,0000 |     0,4444 |
+| LARGE         | =LARGE(TRUE(),1)                                           |          NUM |          NUM |     1,0000 |
+| COUNTA        | =COUNTA(1,)                                                |       2,0000 |       1,0000 |     2,0000 |
+| XNPV          | =XNPV(-0.9, A2:D2, A3:D3)<br>where 2nd and 3rd row: 1,2,3,4 |      10,1272 |  10,12716959 |        NUM |
+| SKEW          | =SKEW(TRUE(),FALSE(),)                                     |       1,7321 |        DIV/0 |     1,7321 |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:109]: I'm not sure if last comma is necessary 🤔 
```suggestion
| SKEW          | =SKEW(TRUE(), FALSE())                                    |       1,7321 |        DIV/0 |     1,7321 |
```

File: docs/guide/known-limitations.md [hunk 5]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| HARMEAN       | =HARMEAN(TRUE(),"4")                                       |       1,6000 |       4,0000 |     1,6000 |
+| GEOMEAN       | =GEOMEAN(TRUE(),"4")                                       |       2,0000 |       4,0000 |     2,0000 |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:111]:
```suggestion
| HARMEAN       | =HARMEAN(TRUE(), "4")                                      |       1,6000 |       4,0000 |     1,6000 |
| GEOMEAN       | =GEOMEAN(TRUE(), "4")                                      |       2,0000 |       4,0000 |     2,0000 |
```

File: docs/guide/known-limitations.md [hunk 6]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| CHISQ.TEST    | =CHISQ.TEST(A1:C2, A1:F1)                                  |          N/A |          N/A |      DIV/0 |
+| BINOM.INV     | =BINOM.INV(1, 0.8, 0.2)                                    |       0,0000 |       1,0000 |     1,0000 |
+| BINOM.INV     | =BINOM.INV(-0.001, 0.5, 0.5)                               |          NUM |       0,0000 |        NUM |
+| BINOM.INV     | =BINOM.INV(10, 0, 0.5)                                     |       0,0000 |          NUM |        NUM |
+| BINOM.INV     | =BINOM.INV(10, 1, 0.5)                                     |      10,0000 |          NUM |        NUM |
+| DEVSQ         | =DEVSQ(A2,A3)                                              |       0,0000 |       0,0000 |        NUM |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:117]:
```suggestion
| DEVSQ         | =DEVSQ(A2, A3)                                             |       0,0000 |       0,0000 |        NUM |
```

File: docs/guide/known-limitations.md [hunk 7]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| HYPGEOM.DIST  | =HYPGEOM.DIST(12.1, 12, 20, 40, TRUE())                    |          NUM |          N/A |     1,0000 |
+| HYPGEOM.DIST  | =HYPGEOM.DIST(12.1, 20, 12, 40, TRUE())                    |          NUM |          N/A |     1,0000 |
+| HYPGEOM.DIST  | =HYPGEOM.DIST(1, 2, 3, 4)                                  |          N/A |       0,5000 |        NUM |
+| HYPGEOM.DIST  | =HYPGEOM.DIST(4, 12, 20, 40, TRUE())                       |       0,1504 |          N/A |     0,1504 |
+| TDIST         | =TDIST(0, 1, 1.5)                                          |          NUM |       0,5000 |     0,5000 |
+| T.INV.2T      | =T.INV.2T(0, 1)                                            |          NUM |          NUM |      DIV/0 |
+| T.DIST        | =T.DIST(1, 0.9, FALSE())                                   |          NUM |          NUM |      DIV/0 |
+| AVEDEV        | =AVEDEV(TRUE(),FALSE(),)                                   |       0,4444 |       0,0000 |     0,4444 |
+| LARGE         | =LARGE(TRUE(),1)                                           |          NUM |          NUM |     1,0000 |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:106]:
```suggestion
| AVEDEV        | =AVEDEV(TRUE(), FALSE())                                  |       0,4444 |       0,0000 |     0,4444 |
| LARGE         | =LARGE(TRUE(), 1)                                          |          NUM |          NUM |     1,0000 |
```

File: docs/guide/known-limitations.md [hunk 8]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+| TBILLEQ       | =TBILLEQ("1/2/2000",<br>"31/1/2001", 0.1)                  |       0,1128 |        VALUE |      VALUE |
+| TBILLEQ       | =TBILLEQ(0, 360, 0.1)                                      |       0,1127 |       0,1097 |     0,1097 |
+| TBILLEQ       | =TBILLEQ(0, 365, 0.1)                                      |       0,1128 |       0,1098 |     0,1098 |
+| GCD           | =GCD(1000000000000000000.0)                                |          NUM |        1E+18 |        NUM |
+| COMBIN        | =COMBIN(1030, 0)                                           |          NUM |          NUM |     1,0000 |
+| RRI           | =RRI(1, -1, -1)                                            |       0,0000 |          NUM |     0,0000 |
+| DAYS          | =DAYS(-1, 0)                                               |          NUM |      -1,0000 |        NUM |
+| DAYS          | =DAYS(0, -1)                                               |          NUM |       1,0000 |        NUM |
+| DATEDIF       | =DATEDIF(-1, 0,   "Y")                                     |          NUM |       0,0000 |        NUM |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:73]:
```suggestion
| DATEDIF       | =DATEDIF(-1, 0, "Y")                                       |          NUM |       0,0000 |        NUM |
```

File: docs/guide/known-limitations.md [hunk 9]
```diff
@@ -34,3 +33,86 @@ you can't compare the arguments in a formula like this:
...
+
+### Built-in function implementation differences
+
+| Function      | Example                                                    | HyperFormula | Google Sheets |     Microsoft Excel |
+|---------------|------------------------------------------------------------|-------------:|-------------:|-----------:|
+| TBILLEQ       | =TBILLEQ(0, 180, 1.9)                                      |      38,5278 |          NUM |        NUM |
+| TBILLEQ       | =TBILLEQ(0, 180, 2)                                        |       0,0000 |          NUM |     0,0000 |
+| TBILLEQ       | =TBILLEQ("1/2/2000",<br>"31/1/2001", 0.1)                  |       0,1128 |        VALUE |      VALUE |
```
Comment (wojciechczerniak) [inline docs/guide/known-limitations.md:65]:
```suggestion
| TBILLEQ       | =TBILLEQ("1/2/2000", "31/1/2001", 0.1)                  |       0,1128 |        VALUE |      VALUE |
```

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: docs/guide/known-limitations.md (hunk 1 above)
Comment (kirszenbaum) [reply docs/guide/known-limitations.md]: @wojciechczerniak Done: 69de5fb

**Review by wojciechczerniak — APPROVED** [review]
👍

**Pull request merged**